File: gracie.lee/src/main/java/homework/chapter_6/homework1/Team.java
package homework.chapter_6.homework1;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "Hw1Team")
@Table(name = "team_hw1")
public class Team {

    @Id
    @GeneratedValue
    @Column(name = "team_id")
    private Long id;

    private String name;

    @OneToMany
    // @JoinColumn(name = "team_id")  ← 주석 해제/처리해서 SQL 차이 확인!
    private List<Member> members = new ArrayList<>();

    protected Team() {
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<Member> getMembers() {
        return members;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: gracie.lee/src/main/kotlin/homework/chapter_6/homework1/Team.kt
package homework.chapter_6.homework1

import jakarta.persistence.*

@Entity(name = "Hw1TeamKt")
@Table(name = "team_hw1_kt")
class Team(
    @Id
    @GeneratedValue
    @Column(name = "team_id")
    val id: Long? = null,

    var name: String? = null,

    @OneToMany
    // @JoinColumn(name = "team_id")  <- 주석 해제/처리해서 SQL 차이 확인!
    val members: MutableList<Member> = mutableListOf()
)
